superset/typing.py

```diff
@@ -16,15 +16,31 @@
 # under the License.
 from typing import Any, Callable, Dict, List, Optional, Tuple, Union
 
-from flask import Flask
+import flask
+import werkzeug
 from flask_caching import Cache
 
-CacheConfig = Union[Callable[[Flask], Cache], Dict[str, Any]]
+CacheConfig = Union[Callable[[flask.Flask], Cache], Dict[str, Any]]
 DbapiDescriptionRow = Tuple[
     str, str, Optional[str], Optional[str], Optional[int], Optional[int], bool
 ]
 DbapiDescription = Union[List[DbapiDescriptionRow], Tuple[DbapiDescriptionRow, ...]]
 DbapiResult = List[Union[List[Any], Tuple[Any, ...]]]
 FilterValue = Union[float, int, str]
 FilterValues = Union[FilterValue, List[FilterValue], Tuple[FilterValue]]
+Granularity = Union[str, Dict[str, Union[str, float]]]
+Metric = Union[Dict[str, str], str]
+QueryObjectDict = Dict[str, Any]
 VizData = Optional[Union[List[Any], Dict[Any, Any]]]
+
+# Flask response.
+Base = Union[bytes, str]
+Status = Union[int, str]
+Headers = Dict[str, Any]
+FlaskResponse = Union[
+    flask.wrappers.Response,
+    werkzeug.wrappers.Response,
+    Base,
+    Union[Base, Status],
+    Union[Base, Status, Headers],
+]
```

Comment:
As we already have a wrapping `Union` here, couldn't this be expressed as `Union[..., Base, Status, Headers]`?

Reply:
@villebro, that's actually a typo as the later `Union` should be a `Tuple` and given that `flask.wrappers.Response` is derived from `werkzeug.wrappers.Response` per [here](https://github.com/pallets/flask/blob/master/src/flask/wrappers.py#L93) this could be:

```python
FlaskResponse = Union[
    werkzeug.wrappers.Response,
    Base,
    Tuple[Base, Status],
    Tuple[Base, Status, Headers],
]
```

Reply:
Ok, that makes sense. Unrelated comment, it's really strange how practically none of the major libraries have implemented type annotations yet. Yes, stubs exist for many, but that doesn't really help when reading code.